web/client/components/Snapshot/SnapshotPanel.jsx

```diff
@@ -0,0 +1,104 @@
+    },
+    getDefaultProps() {
+        return {
+            id: "snapshot_panel",
+            layers: [],
+            snapshot: { state: "DISABLED", img: {}},
+            browser: {},
+            icon: <Glyphicon glyph="camera"/>,
+            onStatusChange: () => {},
+            downloadImg: () => {},
+            saveBtnText: "Save",
+            serviceBoxUrl: null,
+            dateFormat: {day: "numeric", month: "long", year: "numeric"}
+        };
+    },
+    renderLayers() {
+        let items = this.props.layers.map((layer) => {
+            if (layer.visibility) {
+                return (layer.title);
+            }
+        });
+        return items;
+    },
+    localDownload() {
+        return (<Button bsSize="xs" disabled={(this.props.snapshot.state === "SHOTING")}
+            href={this.props.snapshot.img.data.replace(/^data:image\/[^;]/, 'data:application/octet-stream')} download="snapshot.png"
+                            >
+                            <Glyphicon glyph="floppy-save" />{this.props.saveBtnText}
+                            </Button>);
+    },
+    remoteDownload() {
+        return (<Button bsSize="xs" disabled={(this.props.snapshot.state === "SHOTING")}
+                onClick={this.onClick}>
+                <Glyphicon glyph="floppy-save" />{this.props.saveBtnText}
+                </Button>);
+    },
+    renderError() {
+        if (this.props.snapshot.error) {
+            return (<Row style={{marginTop: "5px"}}>
+                    {this.props.snapshot.error}
+                    </Row>);
+        }
+    },
+    render() {
+        return ( (this.props.snapshot.img && this.props.snapshot.img.data) || this.props.snapshot.state === "READY") ? (
+            <Grid header={this.props.name} className="snapshot-panel" fluid={true}>
+                <Row>
+                    <Col xs={7} sm={7} md={7}><Image src={this.props.snapshot.img.data} responsive thumbnail/></Col>
+                    <Col xs={5} sm={5} md={5}>
+                       <Table responsive>
+                            <tbody>
+                               <tr>
+                                <td>Date</td><td> <DateFormat dateParams={this.props.dateFormat}/></td>
+                                </tr>
+                                <tr><td>Layers</td><td>{this.renderLayers()}</td></tr>
+                                <tr><td>Size</td><td>{this.props.snapshot.img.width}X{this.props.snapshot.img.height}</td>
+                                </tr>
+                            </tbody>
+                        </Table>
+                    </Col>
+                </Row>
+
+                <Row className="pull-right" style={{marginTop: "5px"}}>
+                    {(this.props.browser.ie || (this.props.snapshot.img.size > 2000 && this.props.browser.chrome)) ? this.remoteDownload() : this.localDownload()}
+                </Row>
+                {this.renderError()}
+            </Grid>
+        ) : null;
```

Comment:
On first render, the window should show loading messages, not a blank window
![image](https://cloud.githubusercontent.com/assets/1279510/12888166/fca3afc2-ce78-11e5-95ba-87b7255cafe3.png)